slackbridge/bots.py

```diff
@@ -129,17 +131,48 @@ def log(self, method, message):
         return method(full_message)
 
     def signedOn(self):
-        # If already registered, auth in
-        self.msg('NickServ', 'IDENTIFY {}'.format(self.nickserv_password))
-        # And if not, register for the first time
-        self.msg('NickServ', 'GROUP {} {}'.format(self.target_group_nick,
-                                                  self.nickserv_password))
+        self.nickserv_auth()
+
         for channel_name in self.joined_channels:
             self.log(log.msg, 'Joining #{}'.format(channel_name))
             self.join(channel_name)
 
         self.away('Default away for startup.')
 
+    def nickserv_auth(self):
+        # If already registered, authenticate yourself to Nickserv
+        self.msg('NickServ', 'IDENTIFY {}'.format(self.nickserv_password))
+        # And if not, register for the first time,
+        # if the nickname is the intended one.
+        if self.nickname == self.intended_nickname:
+            self.msg('NickServ', 'GROUP {} {}'.format(
+                self.target_group_nick, self.nickserv_password))
+
+    # Called whenever the client wants to set a nickname, such as on startup
+    # or if there's a collision.
+    def setNick(self, nickname):
+        super().setNick(nickname)
+        self.nickserv_auth()
+
+    def nickChanged(self, nick):
+        super().nickChanged(nick)
+        if nick != self.intended_nickname:
+            self.log(log.msg,
+                     'Attempting to change nick to {} in 10 seconds.'.format(
+                         self.intended_nickname))
+            reactor.callLater(10, self.setNick, self.intended_nickname)
+
+    # Called when an attempted nickname change collides with an existing one.
+    def irc_ERR_NICKNAMEINUSE(self, prefix, params):
+        # Appends an underscore to the intended nickname,
+        # and changes the bot's nickname to that
+        super().irc_ERR_NICKNAMEINUSE(prefix, params)
+
+        self.log(log.msg,
+                 'Attempting to change nick to {} in 10 seconds.'.format(
+                     self.intended_nickname))
+        reactor.callLater(10, self.setNick, self.intended_nickname)
```

Comment:
This seems like a pretty close duplicate of `nickChanged`, could they be compressed into one method instead so that the logging and the `callLater` are not duplicated in both places?

Reply:
If a nick is in use it will mean the server will change your nick eventually. This seems unnecessary because we can always count on `nickChanged` being called, which will then call `setNick` via `reactor`.

Reply:
Ah, sounds good then. Removing the function.